SystemTests/roles/robot/files/piezo.robot

```diff
@@ -1,20 +1,57 @@
 ###############################################################################
 *** Settings ***
 Documentation     A test suite with for the Piezo system.
-Resource          s3methods.robot
+Library           Collections
+Library           String
 Resource          k8s_methods.robot
+Resource          requests_helpers.robot
+Resource          s3methods.robot
 
 ###############################################################################
 *** Test Cases ***
-Grafana Is Up
-    Check K8s Route Returns 200 Response    /
-
-Prometheus Is Up
-    Check K8s Route Returns 200 Response    /prometheus/graph
-
-Word Count Example
-    # Check that the correct files are in S3
-    File Should Exist In S3 Bucket    kubernetes   wordcount.py
-    File Should Exist In S3 Bucket    kubernetes   big.txt
-    File Should Not Exist In S3 Bucket    kubernetes  output
-    # TODO
+Grafana Returns Ok Response
+    ${response}=  Get Request From Route   /
+    Confirm Ok Response  ${response}
+
+Prometheus Returns Ok Response
+    ${response}=  Get Request From Route   /prometheus/graph
+    Confirm Ok Response  ${response}
+
+Piezo Heartbeat Returns Ok Response
+    ${response}=  Get Request From Route   /piezo/
+    Confirm Ok Response  ${response}
+    ${data}=    Get Response Data   ${response}
+    ${expected}=    Create Dictionary   running=true
+    Dictionaries Should Be Equal    ${data}   ${expected}
+
+Get Logs Of Non Job Returns Not Found Response
+    ${response}=  Get Logs From Spark Driver    dummy
+    Confirm Not Found Response  ${response}
+
+Delete Job Of Non Job Returns Not Found Response
+    ${response}=    Delete Spark Job    dummy
+    Confirm Not Found Response  ${response}
+
+Submit Spark Pi Job Returns Ok Response
+    ${response}=    Submit SparkPi Job    spark-pi-3f69c
+    Confirm Ok Response  ${response}
+    ${data}=    Get Response Data   ${response}
+    Should Be Equal As Strings    ${data["message"]}    Job driver created successfully
+    Should Be Equal As Strings    ${data["driver_name"]}   spark-pi-3f69c-driver
+
+Can Get Logs Of Submitted Spark Job
+    ${job_name}=     Set Variable   spark-pi-fe244
+    ${driver_name}=   Get Driver Name   ${job_name}
+    Submit SparkPi Job    ${job_name}
+    Sleep   1 minute
+    ${response}=  Get Logs From Spark Driver    ${driver_name}
+    ${joblog}=    Get Response Data Message   ${response}
+    ${pi_lines}=    Get Lines Containing String   ${joblog}   Pi is roughly 3.14
```

Comment:
I would maybe make the string matching slightly less strict. I believe I have seen the basic job return values like 3.139...before. I know it shouldn't but just to be safe

Reply:
cf61868